risks/MASVS-STORAGE/1-store-sensitive-data-securely/data-unencrypted-internal/risk.md

```diff
@@ -0,0 +1,41 @@
+---
+title: Sensitive Data Stored Unencrypted in Internal Locations
+alias: data-unencrypted-internal
+platform: [android, ios]
+profiles: [L2]
+mappings:
+  masvs-v1: [MSTG-STORAGE-2]
+  masvs-v2: [MASVS-STORAGE-1, MASVS-CRYPTO-2]
+  mastg-v1: [MASTG-TEST-0052, MASTG-TEST-0001]
+
+---
+
+## Overview
+
+Mobile apps may need to store sensitive data locally within the application sandbox and this data is at risk of exposure via, for example, incorrect file permissions, an app vulnerability, device vulnerability or data backup mechanisms.
+
+[sensitive data](MASTG-THEORY-0023.md "Sensitive Data") may include personally identifiable information (PII), passwords, cryptographic keys or session tokens.
+
+Sensitive data stored locally on the device should be encrypted, and any keys used for encryption methods should be protected by the device's hardware-backed keystore, where available.
+
+## Impact
+
+- **Loss of Confidentiality**: Under the right conditions an attacker could extract sensitive data stored internally within the application sandbox leading to loss of confidentiality and enable further attacks such as identity theft or account takeover.
+
+## Modes of Introduction
+
+- **Data Stored Unencrypted**: Sensitive data is written to the app's private data directory (sandbox) unencrypted.
+- **Hardcoded Encryption Key**: Sensitive data is encrypted but the key is hardcoded inside the application.
+- **Encryption Key Stored on Filesystem**: Sensitive data is encrypted but the key is stored alongside it or in another easily accessible location.
+- **Encryption Used is Insufficient**: Sensitive data is encrypted but the encryption is not considered to be strong.
+
+## Mitigations
+
+- Avoid storing sensitive data locally at all.
+- Use the platform's hardware-backed keystore solution to store the key used for encryption.
+- Use platform features such as [androidx.security.crypto](https://developer.android.com/jetpack/androidx/releases/security) to safely store files and sharedpreferences.
```

Comment:
This library is now [deprecated](https://developer.android.com/privacy-and-security/cryptography?_gl=1*1vy90y4*_up*MQ..*_ga*NDEwODg2ODY1LjE3MDg5NzAwNTM.*_ga_6HH9YJMN9M*MTcwODk3MDA1My4xLjAuMTcwODk3MDA1My4wLjAuMA..#jetpack_security_crypto_library). I don't think there is alternative just now.

Reply:
Nice spot, thanks! Looks like Android is still recommending it in places (https://developer.android.com/privacy-and-security/security-tips#kotlin) so I'm not sure what to put as an alternative. We could leave it generic "use platform features" since this is a risk which is supposed to be platform agnostic, but I would at least like to link to some official guidance.

Reply:
I've flagged this for the call next week to ask whether the mitigations should be included here per existing examples or a mitigations document per the guide. Depending on the answer this could get removed or updated (if I can find a better recommendation for an Android platform-provided solution to encrypting local data)

Reply:
There's no platform-provided solution for now. The thing a dev could do is to encrypt the data and store the encrypted data using Jetpack DataStore or SharedPreferences. It's basically what EncryptedSharedPreferences does under the hood.

Reply:
Per the call the way forward is to remove specific mitigation information here and refer just to "platform features". Then the mitigations should be contained in their own files so we can link to them and update them with advice such is libraries being deprecated.